src/coreclr/interpreter/compiler.cpp

```diff
@@ -559,8 +559,69 @@ int32_t InterpCompiler::ComputeCodeSize()
     return codeSize;
 }
 
+int32_t InterpCompiler::GetLiveStartOffset(int var)
+{
+    if (m_pVars[var].global)
+    {
+        return 0;
+    }
+    else
+    {
+        return m_pVars[var].liveStart->nativeOffset;
+    }
+}
+
+int32_t InterpCompiler::GetLiveEndOffset(int var)
+{
+    if (m_pVars[var].global)
+    {
+        return m_methodCodeSize;
+    }
+    else
+    {
```

Comment:
The liveEnd is the last instruction that uses this var. This path is not really reached as part of this PR, but I suspect here we would actually want to have the native offset of the following instruction ? At least this is what the convention typically is for other code ranges. I think this would need double checking to see what the runtime API expects

Reply:
You are right, the API is expecting exclusive end offset, the function is implemented here.

https://github.com/dotnet/runtime/blob/e2424d0b988a620f039b6380499ad0f66d1b79ef/src/coreclr/debug/di/module.cpp#L3585